src/main/java/com/bugsnag/Configuration.java

```diff
@@ -97,6 +102,52 @@ public boolean shouldAutoCaptureSessions() {
         return autoCaptureSessions.get();
     }
 
+    /**
+     * Set the endpoints to send data to. By default we'll send error reports to
+     * https://notify.bugsnag.com, and sessions to https://sessions.bugsnag.com, but you can
+     * override this if you are using Bugsnag Enterprise to point to your own Bugsnag endpoint.
+     *
+     * Please note that it is recommended that you set both endpoints. If the notify endpoint is
+     * missing, an exception will be thrown. If the session endpoint is missing, a warning will be
+     * logged and sessions will not be sent automatically.
+     *
+     * Note that if you are setting a custom {@link Delivery}, this method should be called after
+     * the custom implementation has been set.
+     *
+     * @param notify the notify endpoint
+     * @param sessions the sessions endpoint
+     *
+     * @throws IllegalArgumentException if the notify endpoint is empty or null
+     */
+    public void setEndpoints(String notify, String sessions) throws IllegalArgumentException {
+        if (notify == null || "".equals(notify)) {
+            throw new IllegalArgumentException("Notify endpoint cannot be empty or null.");
+        } else {
+            if (delivery instanceof HttpDelivery) {
+                ((HttpDelivery) delivery).setEndpoint(notify);
+            } else {
+                LOGGER.warn("Delivery is not instance of HttpDelivery, cannot set notify endpoint");
+            }
+        }
+
+        boolean invalidSessionsEndpoint = sessions == null || "".equals(sessions);
+        String sessionEndpoint = null;
+
+        if (invalidSessionsEndpoint) {
+            LOGGER.warn("The session tracking endpoint has not been"
+                    + " set. Session tracking is disabled");
+            this.autoCaptureSessions.set(false);
+        } else {
+            sessionEndpoint = sessions;
+        }
+
+        if (sessionDelivery instanceof HttpDelivery) {
+            ((HttpDelivery) sessionDelivery).setEndpoint(sessionEndpoint);
```

Comment:
why does it set the endpoint here when it is possible that we have an invalid sessions endpoint? If we want to do this it might be worth adding a code comment as to why